src/utils.rs

```diff
@@ -75,12 +75,18 @@ impl Estimate {
     }
 
     pub fn record_step(&mut self, value: u64) {
+        self.record_step_with_elapsed_time(value, self.start_time.elapsed())
+    }
+
+    /// This helper function takes an explicit elapsed duration so that it could also be used in
+    /// test code, which must be deterministic.
+    fn record_step_with_elapsed_time(&mut self, value: u64, elapsed: Duration) {
```

Comment:
Is there some way we can test this that does not depend on just extracting a single-caller method here?

Reply:
An alternative option that I have considered was to just add an current time Instant parameter to `record_step()` (instead of calling .elapsed() internally), however I have discarded that approach as it's changing the public API even more. I agree that single-caller helper is not ideal. It's a compromise between deterministic testability and public API churn.

Reply:
Actually, `Estimate` is not part of the public API at all, so I think we can just change whatever we like here.

Reply:
Aha, ok. `pub`-s gave the impression that it is. I'll see what `record_step()` idea looks like.

Reply:
Looks a bit neater. Also improved test_time_per_step() itself a little bit.

Reply:
Let's change `record_step` to take a `now: Instant` argument, instead.